### PR TITLE
more expensive metrics

### PR DESCRIPTION
**Pull request by stefantalpalaru**

### CODE REVIEW

**Comment by dryajov** [comment]
Thanks, I think you might be missing some metrics from mplex.nim.

**Comment by stefantalpalaru** [comment]
I don't want to disable all of them. Just the ones with labels that may change significantly over time, generating too many unique combinations over Prometheus' default retention time of 15 days.

File: libp2p/protocols/pubsub/gossipsub.nim [hunk 1]
```diff
@@ -57,17 +57,18 @@ type
     heartbeatRunning: bool
     heartbeatLock: AsyncLock                   # heartbeat lock to prevent two consecutive concurrent heartbeats
 
-declareGauge(libp2p_gossipsub_peers_per_topic_mesh,
-  "gossipsub peers per topic in mesh",
-  labels = ["topic"])
+when defined(libp2p_expensive_metrics):
```
Comment (arnetheduck) [inline libp2p/protocols/pubsub/gossipsub.nim:60]: these should be added back, they're not expensive - ie the topic list is fairly limited (70 items)

Comment (dryajov) [reply libp2p/protocols/pubsub/gossipsub.nim:60]: yes, it wouldn't hurt, but we can get them back with the define, so might not be a big issue?

Comment (stefantalpalaru) [reply libp2p/protocols/pubsub/gossipsub.nim:60]: > the topic list is fairly limited (70 items)

How many times can the fork version change in 15 days?

**Pull request merged**